scala3doc/src/dotty/dokka/tasty/SyntheticSupport.scala

```diff
@@ -96,13 +96,3 @@ trait SyntheticsSupport:
     given dotc.core.Contexts.Context = qctx.asInstanceOf[scala.quoted.runtime.impl.QuotesImpl].ctx
     val cSym = c.symbol.asInstanceOf[dotc.core.Symbols.Symbol]
     cSym.typeRef.appliedTo(cSym.typeParams.map(_.typeRef)).asInstanceOf[TypeRepr]
-
-  object MatchTypeCase:
-    def unapply(tpe: TypeRepr): Option[(TypeRepr, TypeRepr)] =
-      tpe match
-        case AppliedType(t, Seq(from, to)) /*if t == MatchCaseType*/ =>
```

Comment:
Are these reflect extractor?

Reply:
That was code that we used in the doctool to take apart match types.

Reply:
This indicates that `MatchType` are already fully supported. It is just a matter of knowing what needs to go there. This will need some documentation.